artichoke-backend/src/extn/core/range/range.rb

```diff
@@ -19,6 +19,21 @@ def first(*args)
     ary
   end
 
+  def last(*args)
+    return self.end if args.empty?
+
+    raise ArgumentError, "wrong number of arguments (given #{args.length}, expected 1)" unless args.length == 1
+
+    arg = args[0]
+    raise TypeError, "no implicit conversion of #{arg.class} into Integer" unless arg.respond_to?(:to_int)
```

Comment:
this raise does the same thing with immediate objects like `nil` where it prints the value of `#inspect1 directly.

```
[2.6.3] > (0...1).last(nil)
Traceback (most recent call last):
        5: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /usr/local/var/rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):4
        1: from (irb):4:in `last'
TypeError (no implicit conversion from nil to integer)
```

Reply:
Also for booleans

Reply:
whoops, totally miss that. good call out. thanks.